JabRef 3.0 can't connect to libreoffice

I use ubuntu 14.04 (64 bit) libreoffice 4.2 and the new version of jabref. 

Neither manually pointing to the correct pathes nor automatic connection does work. Below you can see the output of my error console.
```
INFORMATION: Opening: /home/felix/Dokumente/UniDaten/backup_version_6/backup/publikationen felix.bib
Exception in thread "Spin-0" java.lang.NoClassDefFoundError: spin/off/AWTReflectDispatcherFactory$1
	at spin.off.AWTReflectDispatcherFactory$AWTReflectDispatcher.stop(AWTReflectDispatcherFactory.java:102)
	at spin.off.SpinOffEvaluator$1.run(SpinOffEvaluator.java:110)
	at java.lang.Thread.run(Thread.java:745)
Caused by: java.lang.ClassNotFoundException: spin.off.AWTReflectDispatcherFactory$1
	at java.net.URLClassLoader.findClass(URLClassLoader.java:381)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:424)
	at sun.misc.Launcher$AppClassLoader.loadClass(Launcher.java:331)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:357)
	... 3 more
java.lang.NoClassDefFoundError: com/sun/star/comp/helper/Bootstrap
	at net.sf.jabref.openoffice.OOBibBase.simpleBootstrap(OOBibBase.java:183)
	at net.sf.jabref.openoffice.OOBibBase.<init>(OOBibBase.java:99)
	at net.sf.jabref.openoffice.OpenOfficePanel.run(OpenOfficePanel.java:614)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:497)
	at spin.Invocation.evaluate(Invocation.java:175)
	at spin.off.SpinOffEvaluator$1.run(SpinOffEvaluator.java:108)
	at java.lang.Thread.run(Thread.java:745)
Caused by: java.lang.ClassNotFoundException: com.sun.star.comp.helper.Bootstrap
	at java.net.URLClassLoader.findClass(URLClassLoader.java:381)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:424)
	at sun.misc.Launcher$AppClassLoader.loadClass(Launcher.java:331)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:357)
	... 10 more
Dez 16, 2015 12:16:25 PM net.sf.jabref.logic.l10n.Localization translate
WARNUNG: Warning: could not get message translation for "Could not connect to running OpenOffice.
Make sure you have installed OpenOffice with Java support.
If connecting manually, please verify program and library paths.

Error message: com/sun/star/comp/helper/Bootstrap" for locale de
```

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -39,6 +39,7 @@ to [sourceforge feature requests](https://sourceforge.net/p/jabref/features/) by
 - Fixed #443/#445: Fixed sorting and moving special field columns
 - Fixed #498: non-working legacy PDF/PS column removed
 - Fixed #473: Import/export to external database works again
+- Fixed #526: OpenOffice/LibreOffice connection works again on Linux/OSX
 
 ### Removed
 - Removed file history size preference (never available from the UI)

--- a/src/main/java/net/sf/jabref/openoffice/OOBibBase.java
+++ b/src/main/java/net/sf/jabref/openoffice/OOBibBase.java
@@ -164,8 +164,9 @@ class OOBibBase {
 
         ClassLoader loader = ClassLoader.getSystemClassLoader();
         if (loader instanceof URLClassLoader) {
-            try (URLClassLoader cl = (URLClassLoader) loader) {
-                Class<URLClassLoader> sysclass = URLClassLoader.class;
+            URLClassLoader cl = (URLClassLoader) loader;
+            Class<URLClassLoader> sysclass = URLClassLoader.class;
+            try {
 
                 Method method = sysclass.getDeclaredMethod("addURL", URL.class);
                 method.setAccessible(true);

--- a/src/main/java/net/sf/jabref/openoffice/OpenOfficePanel.java
+++ b/src/main/java/net/sf/jabref/openoffice/OpenOfficePanel.java
@@ -539,13 +539,13 @@ public class OpenOfficePanel extends AbstractWorker implements PushToApplication
             }
             else if (OS.OS_X) {
                 sOffice = ooPath + "/Contents/MacOS/soffice.bin";
-                ooBaseDirectory = ooPath + "/Contents/basis-link/ure-link/share/java";
+                ooBaseDirectory = ooPath + "/Contents/basis-link/program/classes";
                 unoilDir = ooPath + "/Contents/basis-link/program/classes";
             }
             else {
                 // Linux:
                 unoilDir = ooJars + "/program/classes";
-                ooBaseDirectory = ooJars + "/ure-link/share/java";
+                ooBaseDirectory = ooJars + "/program/classes";
             }
         }
 

